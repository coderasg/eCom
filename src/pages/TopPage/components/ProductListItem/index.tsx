import React from 'react';
import { Product } from 'Products';
import { Link } from 'react-router-dom';

import ProductImage from './../../../../components/ProductImage';
import Like from './../../../../components/Like';

import style from './productListItem.module.scss';

type Props = {
  data: Product;
};

function ProductListItem(props: Props) {
  const { data } = props;


  return (
    <div data-testid="ProductListItem-wrapper" className={style.productListItem}>
      <Link to={`${data.id}`} >
        <ProductImage src={data.image}  bannerText={data.is_sold_out ? 'SOLD' : ''}/>
        <section className={style.productSummary}>
          <div>{data.name}</div>
          <div className={style.price}>
            <span>{`Ò°${data.price}`}</span>
            <Like count={data.like_count} />
          </div>
        </section>
      </Link>
    </div>
  );
}

export default ProductListItem;
