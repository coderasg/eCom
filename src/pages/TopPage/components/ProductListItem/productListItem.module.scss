@use './../../../../variables';

.productListItem {
  text-overflow: ellipsis;
  flex: 0 0 25%;
  box-sizing: border-box;
  margin: 1.25%;
  background-color: white;
  color: #5f5f5f;
  max-width: 400px;

  .productSummary {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 5px;
    min-height: 50px;

    .price {
      font-weight: bold;
      margin-top: auto;
      display: flex;
      justify-content: space-between;
    }
  }

  a {
    text-decoration: none;
    color: black;
  }
}

@media (max-width: 480px) {
  .productListItem {
    flex: 1 0 45%;
  }
}