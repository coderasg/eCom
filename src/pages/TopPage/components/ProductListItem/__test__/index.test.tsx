import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import ProductListItem from '../index';

describe('(Component) ProductListItem', () => {
  const defaultProps = {
    data: {
      id: 1,
      name: 'Light pink shoes',
      description: 'Pre-owned,',
      like_count: 91,
      comment_count: 59,
      price: 51,
      is_sold_out: false,
      shipping_fee: '送料込み',
      image: 'images/image_1.png',
      category_id: 2,
    },
  };
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<BrowserRouter><ProductListItem {...defaultProps} /></BrowserRouter>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
