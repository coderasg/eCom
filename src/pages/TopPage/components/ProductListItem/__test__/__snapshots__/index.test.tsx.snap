// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductListItem should pass snapshot test 1`] = `
<div
  data-testid="ProductListItem-wrapper"
>
  <a
    href="/1"
    onClick={[Function]}
  >
    <div
      className="undefined undefined"
    >
      <img
        data-testid="ProductImage-wrapper"
        src="images/image_1.png"
      />
    </div>
    <section>
      <div>
        Light pink shoes
      </div>
      <div>
        <span>
          Ò°51
        </span>
        <span
          data-testid="Like-wrapper"
        >
          <SvgIcon
            height={15}
            width={15}
          />
          <span>
            91
          </span>
        </span>
      </div>
    </section>
  </a>
</div>
`;
